src/diff_generate.c

```diff
@@ -564,14 +564,14 @@ int git_diff__oid_for_file(
 {
 	git_index_entry entry;
 
-	if (!git__is_sizet(size)) {
+	if (size < 0 || size > UINT32_MAX) {
```

Comment:
This can be squashed into the previous commit where you introduce the `git__is_sizet` check.

Reply:
Sure.